benchmarks/benchmark_serving.py

```diff
@@ -167,27 +161,104 @@ def sample_requests(
   )  # adjust this code based on tokenizer method
   tokenized_dataset = []
   for i in range(len(dataset)):
-    output_len = len(completion_token_ids[i])
-    tokenized_dataset.append((prompts[i], prompt_token_ids[i], completions[i], output_len))
+    prompt_len = len(prompt_token_ids[i])
+    completion_len = len(completion_token_ids[i])
+    tokenized_dataset.append(
+      (prompts[i], prompt_token_ids[i], completions[i], prompt_len, completion_len)
+    )
 
   # Filter out too long sequences.
   filtered_dataset: List[InputRequest] = []
 
-  for prompt, prompt_token_ids, output, output_len in tokenized_dataset:
-    prompt_len = len(prompt_token_ids)
-    if prompt_len < 4 or output_len < 4:
+  for prompt, prompt_token_ids, completion, prompt_len, completion_len in tokenized_dataset:
+    if prompt_len < 4 or completion_len < 4:
       # Prune too short sequences.
       # This is because TGI causes errors when the input or output length
       # is too short.
       continue
+    if prompt_len > 1024 or prompt_len + completion_len > 2048:
+      # Prune too long sequences.
+      continue
+    request = InputRequest(prompt, prompt_len, completion, max_output_length or completion_len)
+    filtered_dataset.append(request)
+
+  if max_output_length is None:
+    print("In InputRequest, pass in actual output_length for each sample")
+  else:
+    print(f"In InputRequest, pass in max_output_length: {max_output_length} for each sample")
+
+  print(f"The dataset contains {len(tokenized_dataset)} samples.")
+  print(f"The filtered dataset contains {len(filtered_dataset)} samples.")
+
+  return filtered_dataset
+
+
+def load_openorca_dataset(
+    dataset_path: str,
+    tokenizer: Any,
+    max_output_length: Optional[int] = None,
+) -> List[InputRequest]:
+
+  # Load the dataset.
+  with open(dataset_path) as f:
+    dataset = json.load(f)
+
+  # Tokenize the prompts and completions.
+  prompts = dataset["prompts"]
+  outputs = dataset["results"]
+  n = len(prompts)
+  prompt_token_ids = tokenizer.tokenize(prompts)
+  output_token_ids = tokenizer.tokenize(outputs)
```

Comment:
I feel we could extract this part out as a func for different dataset, and the rest are identical and thus we could keep them in the sample_request function?

Reply:
Discussed offline, there are some existing data processing that may not be necessary. Will revisit and refactor the data preprocessing part if needed.